### PR TITLE
Adding hal.interface.workgroup.* ops.

### PR DESCRIPTION
**Pull request by benvanik**
hal.interface.workgroup.size will always return 1,1,1 at runtime today (until that is available) but a pass is added to propagate it as constant data when present on entry points.

### CODE REVIEW

File: iree/compiler/Conversion/LinalgToLLVM/ConvertToLLVM.cpp [hunk 1]
```diff
@@ -132,6 +132,34 @@ class RemoveMakeRankedShape : public ConvertToLLVMPattern {
...
+                                         LLVMTypeConverter &typeConverter)
+      : ConvertToLLVMPattern(Op::getOperationName(), context, typeConverter) {}
+
+  LogicalResult matchAndRewrite(
+      Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    auto newFuncOp = cast<LLVM::LLVMFuncOp>(rewriter.getBlock()->getParentOp());
+    auto xyzTy =
+        LLVM::LLVMType::getInt32Ty(rewriter.getContext()).getPointerTo();
+    auto xyzArgument = newFuncOp.getArgument(ArgIndex);
```
Comment (asaadaldien) [inline iree/compiler/Conversion/LinalgToLLVM/ConvertToLLVM.cpp:148]: Nice! this might be the only valid way to do this kind of conversion `func() -> llvm.func(%fixed_args....)`.

**Comment by asaadaldien** [comment]
Looks good! just few tests to update
https://github.com/google/iree/tree/main/iree/compiler/Conversion/LinalgToLLVM/test

File: iree/compiler/Dialect/HAL/IR/test/interface_ops.mlir [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  %1 = hal.interface.workgroup.count[1] : index
+  // CHECK: %workgroup_size_z = hal.interface.workgroup.size[2] : index
+  %2 = hal.interface.workgroup.size[2] : index
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @interface_io_tensors
+func @interface_io_tensors() {
```
Comment (asaadaldien) [inline iree/compiler/Dialect/HAL/IR/test/interface_ops.mlir:17]: Nit: Wanna move these tests to a different commit in your stack ?

**Review by asaadaldien — APPROVED** [review]

File: iree/compiler/Dialect/HAL/IR/test/interface_ops.mlir (hunk 2 above)
Comment (benvanik) [reply iree/compiler/Dialect/HAL/IR/test/interface_ops.mlir:17]: ah this is just moving them while touching the file - git absorb issues probably :)
